vdyp-vri/src/test/java/ca/bc/gov/nrs/vdyp/vri/VriStart.java

```diff
@@ -0,0 +1,235 @@
+		} catch (NoSuchElementException ex) {
+			throw validationError("Species file has fewer records than polygon file.", ex);
+		}
+
+		log.trace("Getting sites for polygon {}", polygon.getPolygonIdentifier());
+		Collection<VriSite> sites;
+		try {
+			sites = siteStream.next();
+		} catch (NoSuchElementException ex) {
+			throw validationError("Sites file has fewer records than polygon file.", ex);
+		}
+
+		// Validate that layers belong to the correct polygon
+		for (var layer : layers.values()) {
+			if (!layer.getPolygonIdentifier().equals(polygon.getPolygonIdentifier())) {
+				throw validationError(
+						"Record in layer file contains layer for polygon %s when expecting one for %s.",
+						layer.getPolygonIdentifier(), polygon.getPolygonIdentifier()
+				);
+			}
+			layer.setSpecies(new HashMap<>());
+		}
+
+		for (var spec : species) {
+			var layer = layers.get(spec.getLayer());
+			// Validate that species belong to the correct polygon
+			if (!spec.getPolygonIdentifier().equals(polygon.getPolygonIdentifier())) {
+				throw validationError(
+						"Record in species file contains species for polygon %s when expecting one for %s.",
+						layer.getPolygonIdentifier(), polygon.getPolygonIdentifier()
+				);
+			}
+			if (Objects.isNull(layer)) {
+				throw validationError(
+						"Species entry references layer %s of polygon %s but it is not present.", layer,
+						polygon.getPolygonIdentifier()
+				);
+			}
+			layer.getSpecies().put(spec.getGenus(), spec);
+		}
+
+		for (var site : sites) {
+			var layer = layers.get(site.getLayer());
+			// Validate that species belong to the correct polygon
+			if (!site.getPolygonIdentifier().equals(polygon.getPolygonIdentifier())) {
+				throw validationError(
+						"Record in site file contains site for polygon %s when expecting one for %s.",
+						layer.getPolygonIdentifier(), polygon.getPolygonIdentifier()
+				);
+			}
+			if (Objects.isNull(layer)) {
+				throw validationError(
+						"Site entry references layer %s of polygon %s but it is not present.", layer,
+						polygon.getPolygonIdentifier()
+				);
+			}
+			layer.getSites().put(site.getSiteGenus(), site);
+		}
+
+		polygon.setLayers(layers);
+
+		return polygon;
+
+	}
+
+	static final EnumSet<PolygonMode> ACCEPTABLE_MODES = EnumSet.of(PolygonMode.START, PolygonMode.YOUNG);
+
+	Optional<VdypPolygon> processPolygon(int polygonsRead, VriPolygon polygon) throws ProcessingException {
+		return Optional.empty();
+
+	}
+
+	VdypPolygon createVdypPolygon(VriPolygon sourcePolygon, Map<LayerType, VdypLayer> processedLayers)
+			throws ProcessingException {
+
+		// TODO expand this
+
+		var vdypPolygon = VdypPolygon.build(builder -> builder.copy(sourcePolygon, x -> x.get()));
+		vdypPolygon.setLayers(processedLayers);
+		return vdypPolygon;
+	}
+
+	@Override
+	public VdypApplicationIdentifier getId() {
+		return VdypApplicationIdentifier.VRI_START;
+	}
+
+	@Override
+	protected BaseControlParser getControlFileParser() {
```

Comment:
Is it intended that this returns null, or have you just not gotten to writing an implementation?